packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -244,6 +245,72 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     return validSubmission;
   }, [setError]);
 
+  const handleVisitAttributes = useCallback(
+    async (visitAttributes: { [p: string]: string }, visitUuid: string) => {
+      const existingAttributes = visitToEdit?.attributes?.map((attribute) => attribute.attributeType.uuid) || [];
+      const attributes = Object.entries(visitAttributes).map(([key, value]) => ({
+        attributeType: key,
+        value: value as string,
+      }));
+
+      for (const attribute of attributes) {
+        if (attribute.attributeType && existingAttributes.includes(attribute.attributeType)) {
+          const attributeToEdit = visitToEdit.attributes.filter(
```

Comment:
Better because this short-circuits iterating across the whole array:

```suggestion
          const attributeToEdit = visitToEdit.attributes.find(
```